packages/composite-checkout/src/components/checkout-steps.tsx

```diff
@@ -564,6 +566,7 @@ export function CheckoutFormSubmit( {
 		<SubmitButtonWrapper className="checkout-steps__submit-button-wrapper" ref={ submitWrapperRef }>
 			{ submitButtonHeader || null }
 			<CheckoutSubmitButton
+				validateForm={ validateForm }
```

Comment:
I'm not totally against this approach, but I wonder if it's necessary since the package already has support for validation in the `isCompleteCallback` prop of each `CheckoutStep` and the `useFormStatus()` hook. I worry that adding this prop might add confusion about where validation is designed to be performed, although I can see the flexibility it offers.

🤔 I guess the question is: can this behavior be implemented with the existing system? One way would be to put the `ThirdPartyDevsAccount` inside a `CheckoutStep` just above the button. Another option would be to access `useFormStatus()` directly where you're keeping track of the checkbox validation state and set the form's status to validating which I believe should disable the button automatically.

Reply:
Thanks for your comment, I thought about using `CheckoutStep` but I discarded it as it seemed it involved more than what I required. But I totally agree about using existing functionality so I will iterate over it 👍

Reply:
The more I consider this, the more I think your approach here may be the right one after all.

Using a `CheckoutStep` would be elegant in some ways but it would add a lot more clicks for the user because it would cause the payment method step to have a "Continue" button. The design where the checkbox is basically attached to the submit button definitely feels more correct.

With `useFormStatus` or `disableSubmitButton`, we could disable the button when the checkbox is unchecked, but I better understand what you did here: disabled buttons provide no feedback about what the user needs to do and an error message is better. By adding `validateForm`, we can provide that error message.

I was worried that `validateForm` might expand the API of the (relatively simple) `CheckoutFormSubmit`/`CheckoutSubmitButton`, but those components already provide opportunities for customization and I think this additional function does just that.

So I think you can go ahead with this change, but please add it to the [README](https://github.com/Automattic/wp-calypso/blob/91cdda4454290ede3010d36660d83761225ec2c8/packages/composite-checkout/README.md?plain=1#L96) for the `composite-checkout` package (it looks like none of the props are documented actually 😅 so this would be a good opportunity) and we might want to add tests to prove that it blocks submission, perhaps [around here](https://github.com/Automattic/wp-calypso/blob/91cdda4454290ede3010d36660d83761225ec2c8/packages/composite-checkout/test/checkout.js#L533). I'm happy to help with these parts and could push them to this branch if you'd like.

Reply:
I agree, after trying `CheckoutStep` I was amazed by how simple and powerful is. But then I realised that the user experience changes a bit and the checkout process would increase in one more step.

So I will go ahead with the change and add some documentation and tests. Feel free to add any tests you may consider.

Reply:
Hi @sirbrillig, I have added some tests as part of 0793480e661b5c7ebabe86cdee6b70f83c5e8d10, but I was not sure how to use [existing test renderers](https://github.com/Automattic/wp-calypso/blob/91cdda4454290ede3010d36660d83761225ec2c8/packages/composite-checkout/test/checkout.js#L199), so I have created a new `describe` section with similar functionality to the one [you suggested](https://github.com/Automattic/wp-calypso/blob/91cdda4454290ede3010d36660d83761225ec2c8/packages/composite-checkout/test/checkout.js#L533). 

Please, let me know or feel free to amend it if you think test functionality can be re-used. I was also not sure how to assert  the click of the `Submit` button so I have used `expect( getByText( 'Pay Please' ) ).toBeDisabled();` and the negated one. Again, let me know (or feel free to update it) if you think it could be better _asserted_. Thanks!